pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -29,8 +37,72 @@ func computeDesiredFleetSize(fas *stablev1alpha1.FleetAutoscaler, f *stablev1alp
 	switch fas.Spec.Policy.Type {
 	case stablev1alpha1.BufferPolicyType:
 		return applyBufferPolicy(fas.Spec.Policy.Buffer, f)
+	case stablev1alpha1.WebhookPolicyType:
+		return applyWebhookPolicy(fas.Spec.Policy.Webhook, f)
 	}
 
+	return f.Status.Replicas, false, fmt.Errorf("Wrong Policy Type")
+}
+
+func applyWebhookPolicy(w *stablev1alpha1.WebhookPolicy, f *stablev1alpha1.Fleet) (int32, bool, error) {
+	faReq := v1alpha1.FleetAutoscaleReview{
+		Request: &v1alpha1.FleetAutoscaleRequest{
+			UID:       uuid.NewUUID(),
+			Name:      f.Name,
+			Namespace: f.Namespace,
+			Status:    f.Status,
+		},
+		Response: nil,
+	}
+	b, err := json.Marshal(faReq)
+	url := ""
+	if w.URL != nil {
+		url = *w.URL
+	}
+	var faResp v1alpha1.FleetAutoscaleReview
+	servicePath := ""
+	if w.Service != nil {
+		if w.Service.Path != nil {
+			servicePath = *w.Service.Path
+		}
+		if err != nil {
+			return f.Status.Replicas, false, err
+		}
+
+		if w.Service.Namespace == "" {
+			w.Service.Namespace = "default"
+		}
+		url = fmt.Sprintf("http://%s.%s.svc:8000/%s", w.Service.Name, w.Service.Namespace, servicePath)
+	}
+	if url == "" {
+		return f.Status.Replicas, false, fmt.Errorf("URL was not provided")
+	}
+	c := &http.Client{
```

Comment:
You could reuse the same client.

Reply:
I will used singleton, cause this function has no receiver